text/0000-remove-shell-processes.md

```diff
@@ -0,0 +1,470 @@
+- The difference between a direct and shell process
+- The nuances of argument handling in direct vs. Bash vs. Cmd cases
+- That buildpack-provided profile scripts will not apply to direct processes
+- That a user-provided `.profile` script will not apply to a direct processes
+
+ Removing this special behavior and requiring buildpacks to explicitly include any required shell in the command itself creates a simpler, more comprehensible model for buildpack authors and end users alike.
+
+# What it is
+[what-it-is]: #what-it-is
+
+## Process Schema
+The existing schema for the processes table in `launch.toml` is
+```
+# Old Schema
+
+[[processes]]
+type = "required"
+command = "required"
+args = ["optional"]
+direct = false
+default = false
+```
+
+The new proposed schema is:
+```
+# New Schema
+
+[[processes]]
+type = "required"
+command = ["required"]
+args = ["optional"]
+default = false
+```
+The following changes have been made:
+- **`direct` has been removed** - all processes are executed directly. If `bash` or `cmd.exe` is required it should be included in `command`. No surprises.
+- **`command` is now an array** - Arguments in `command` *will not* be overwritten if a user provides additional arguments at runtime. `args` are default arguments that *will* be overwritten if a user provides additional arguments at runtime. The makes `command` analogous to `Entrypoint` in the OCI spec and `command` in a Kubernetes PodSpec. `args` is analogous to `cmd` and `args` in docker and Kubernetes respectively.
+
+## Using Environment Variables in a Process
+
+One upside to our previous execution strategy was that it enable users to include environment variable references in arguments that were later evaluated in the container. To preserve this feature we can instead adopt the Kubernetes strategy for [environment variables interpolation](https://kubernetes.io/docs/tasks/inject-data-application/define-environment-variable-container/#using-environment-variables-inside-of-your-config). If a buildpack or user includes `$(<env>)` in the `command` or `args` and `<env>` is the name of an environment variable set in the launch environment, the launcher will replace this string with the value of the environment variable after apply buildpack-provided env modifications and before launching the process.
+
+
+# How it Works
+[how-it-works]: #how-it-works
+
+## Buildpack-provided process types
+### Example 1 - A Shell Process
+The Paketo .Net Execute Buildpack may generates shell processes similar to the following:
+```
+[[processes]]
+type = "web"
+command = "dotnet my-app.dll --urls http://0.0.0.0:${PORT:-8080}"
+direct = false
+```
+NOTE: the buildpack API used by this buildpack (`0.5`) predates the introduction of `default`.
+
+Using the new API this process could look like:
+```
+[[processes]]
+type = "web"
+command = ["dotnet", "my-app.dll", "--urls", "http://0.0.0.0:${PORT}"] # the default value of PORT would need to be provided in a layer
```

Comment:
```suggestion
command = ["dotnet", "my-app.dll", "--urls", "http://0.0.0.0:$(PORT)"] # the default value of PORT would need to be provided in a layer
```

Should this follow the k8s interpolation format since this is no longer a shell process?

Reply:
@ekcasey please take a look.